src/server/controllers/changePassword.ts

```diff
@@ -95,6 +95,12 @@ const oktaIdxApiPasswordHandler = async ({
 					introspectResponse,
 					'enroll-authenticator',
 				);
+			} else {
+				// if we're setting a password, we should be in the reset-authenticator remediation
+				validateIntrospectRemediation(
+					introspectResponse,
+					'reset-authenticator',
+				);
```

Comment:
```js
validateIntrospectRemediation(
	introspectResponse,
	path === '/welcome' ? 'enroll-authenticator' : 'reset-authenticator',
);
```

The functions are getting pretty massive and hard to keep it all in my head, might make things easier if we reduced the size of the branches?

non blocking

Reply:
i might have to more refactors before the next feature is implemented, so i'll keep this in mind!

Reply:
Gotcha!